### PR TITLE
Add tags field to country note add

### PR DESCRIPTION
**Pull request by raysonkoh**
## Description

To be merged after #235 

Most of the changes are in `CountryNoteAddCommandParser`. Copied implementation of parsing tags from `client note add`.

## Testing

Added additional tests for `CountryNoteAddCommandParser`. Basically just to check that tags are optional and the other params are compulsory.

Manually tested and checked json storage and GUI. Both are updating correctly.

## Remarks

Will move to `assertParseSuccess` and `assertParseSuccess` in a future issue because most of the country-related files did not use it and it would be easier to change it in one shot.

### CODE REVIEW

**Review by rtshkmr — COMMENTED** [review]
LGTM for me, tests are nicely written. 
I might have missed out small nits, so gonna let another set of eyes see it before either of us approve :safety_vest:

File: src/test/java/seedu/address/logic/parser/CountryNoteAddCommandParserTest.java
Comment (qwoprocks) [inline src/test/java/seedu/address/logic/parser/CountryNoteAddCommandParserTest.java]:
```suggestion
        assertThrows(ParseException.class, () -> parser.parse(" c/MY nt/ t/a"));
        assertThrows(ParseException.class, () -> parser.parse(" nt/ c/MY t/a"));
```
Better isolation for testing no notes

File: src/test/java/seedu/address/logic/parser/CountryNoteAddCommandParserTest.java
Comment (qwoprocks) [inline src/test/java/seedu/address/logic/parser/CountryNoteAddCommandParserTest.java]:
```suggestion
        assertThrows(ParseException.class, () -> parser.parse(" c/RUS t/a nt/random string"));
        assertThrows(ParseException.class, () -> parser.parse(" c/R t/a nt/random string"));
```
Same reason as below, also, should have a noCountryHasNote test method

File: src/test/java/seedu/address/logic/parser/CountryNoteAddCommandParserTest.java
Comment (qwoprocks) [inline src/test/java/seedu/address/logic/parser/CountryNoteAddCommandParserTest.java:70]: Should this be in a separate test function?

File: src/main/java/seedu/address/logic/parser/CountryNoteAddCommandParser.java
Comment (qwoprocks) [inline src/main/java/seedu/address/logic/parser/CountryNoteAddCommandParser.java:55]: tagNoteMap not being updated after setting the tags?

**Review by qwoprocks — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/parser/CountryNoteAddCommandParser.java
Comment (raysonkoh) [reply src/main/java/seedu/address/logic/parser/CountryNoteAddCommandParser.java:55]: Not sure about this since I followed the implementation of `client note add` command parser. @rtshkmr any comments?

File: src/test/java/seedu/address/logic/parser/CountryNoteAddCommandParserTest.java
Comment (raysonkoh) [reply src/test/java/seedu/address/logic/parser/CountryNoteAddCommandParserTest.java:70]: Yup, think so.

File: src/main/java/seedu/address/logic/parser/CountryNoteAddCommandParser.java
Comment (rtshkmr) [reply src/main/java/seedu/address/logic/parser/CountryNoteAddCommandParser.java:55]: ref line 60 

image


the command#execute is the one doing it

Comment (raysonkoh) [reply src/main/java/seedu/address/logic/parser/CountryNoteAddCommandParser.java:55]: I see, can you help to check the latest commit in `CountryNoteAddCommand#execute` whether I am doing it correctly?

Comment (qwoprocks) [reply src/main/java/seedu/address/logic/parser/CountryNoteAddCommandParser.java:55]: @raysonkoh I think we can probably remove the model.updateTagNoteMapWithNote() and instead place it inside the function model.addCountryNote, since addClientNote is doing that as well. Then we can remove model.updateTagNoteMapWithNote() from the execute() function of both client note add and country note add, and make the method private in model

Comment (raysonkoh) [reply src/main/java/seedu/address/logic/parser/CountryNoteAddCommandParser.java:55]: Ok removed calls to `updateTagNoteMapWithNote` in `ClientNoteAddCommand` and `CountryNoteAddCommand`. Now the method will be called inside `ModelManager#addCountryNote` / `ModelManager#addClientNote`.

File: src/main/java/seedu/address/model/ModelManager.java
Comment (qwoprocks) [inline src/main/java/seedu/address/model/ModelManager.java:169]: there's 2 updateTagNoteMapWithNote here

**Review by qwoprocks — APPROVED** [review]
Small issue, otherwise LGTM

File: src/main/java/seedu/address/model/ModelManager.java
Comment (raysonkoh) [reply src/main/java/seedu/address/model/ModelManager.java:169]: I must be blind

**Pull request merged**
